src/include/planner/analyze_plan.h

```diff
@@ -37,19 +38,22 @@ class AnalyzePlan : public AbstractPlan {
 
   explicit AnalyzePlan(storage::DataTable *table);
 
-  explicit AnalyzePlan(std::string table_name);
+  explicit AnalyzePlan(std::string table_name,
+                       concurrency::Transaction *txn = nullptr);
```

Comment:
Yes. Same comments as my previous one. We should not allow default parameter value.